### PR TITLE
Fix disposing DB with open batch

### PR DESCRIPTION
**Pull request by LukaszRozmej**
## Changes:
ReleaseResources before setting disposed flag.
Only touch managed objects when disposing

## Types of changes

What types of changes does your code introduce?
_Put an `x` in the boxes that apply_

- [x] Bugfix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Documentation Update
- [ ] Code style update (formatting, renaming)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] Other (please describe): 

## Testing
**Requires testing**

- [x] Yes
- [ ] No

**In case you checked yes, did you write tests??**

- [x] Yes
- [ ] No

### CODE REVIEW

File: src/Nethermind/Nethermind.Db.Rocks/DbOnTheRocks.cs [hunk 1]
```diff
@@ -525,29 +525,23 @@ private void ReleaseUnmanagedResources()
...
     private void Dispose(bool disposing)
     {
         if (!_isDisposed)
         {
-            // ReSharper disable once ConstantConditionalAccessQualifier
-            _logger?.Info($"Disposing DB {Name}");
-
             if (disposing)
             {
+                if (_logger.IsInfo) _logger.Info($"Disposing DB {Name}");
```
Comment (flcl42) [inline src/Nethermind/Nethermind.Db.Rocks/DbOnTheRocks.cs:538]: Dispose(false) does nothing, right? Could be simplified if so

**Review by rubo — APPROVED** [review]

File: src/Nethermind/Nethermind.Db.Rocks/DbOnTheRocks.cs (hunk 1 above)
Comment (LukaszRozmej) [reply src/Nethermind/Nethermind.Db.Rocks/DbOnTheRocks.cs:538]: Hm... I wonder because the `_rocksBatch.Dispose();` might do useful thing, WDYT?

Comment (flcl42) [reply src/Nethermind/Nethermind.Db.Rocks/DbOnTheRocks.cs:538]: Yep, let batches be disposed, I meant we have Dispose(bool disposing), if `false` we just reset disposing flag, looks useless a bit

Comment (flcl42) [reply src/Nethermind/Nethermind.Db.Rocks/DbOnTheRocks.cs:538]: On other side finalizer may help to raise some useful exceptions if we forget to dispose

**Review by flcl42 — APPROVED** [review]

File: src/Nethermind/Nethermind.Db.Rocks/DbOnTheRocks.cs (hunk 1 above)
Comment (LukaszRozmej) [reply src/Nethermind/Nethermind.Db.Rocks/DbOnTheRocks.cs:538]: So maybe either on !disposing still dispose inner batches? Or add finalizers to batches? Which is better?

Comment (flcl42) [reply src/Nethermind/Nethermind.Db.Rocks/DbOnTheRocks.cs:538]: `Dispose` is cool imho

Comment (flcl42) [reply src/Nethermind/Nethermind.Db.Rocks/DbOnTheRocks.cs:538]: `~DbOnTheRocks` should not even ever be called if we control the lifetime properly. If `!disposing && !_isDisposed` we can notify of a problem and still try to clean up, yes.

Comment (LukaszRozmej) [reply src/Nethermind/Nethermind.Db.Rocks/DbOnTheRocks.cs]: `~DbOnTheRocks` doesn't make sense, the wrapper underneath should have finalizer. The one with native IntPtr, not this one.

Comment (flcl42) [reply src/Nethermind/Nethermind.Db.Rocks/DbOnTheRocks.cs]: `Dispose(false)` is never used now

**Review by flcl42 — APPROVED** [review]

**Review by kamilchodola — APPROVED** [review]

**Review by MarekM25 — APPROVED** [review]

**Pull request merged**